### DISCUSSION TITLE
Missing CSS and JS Files in Nuxt.js Build

### DISCUSSION THREAD

**Discussion by Michael0520** [nuxt/nuxt#25092] [Questions]
## Intro
- Enhancing Content Marketing Efficiency: To streamline rapid updates and content optimization, our content marketing team has adopted WordPress for easy editing and testing.
- Improved HTML Structure: To facilitate content review and editing, we use the Prettier tool to format HTML files, ensuring readability and support for various languages.
- Cost-Efficient Maintenance: Given frequent content updates, our goal is to simplify the workflow for the content team by eliminating hash values from filenames during the build process, reducing complexity.
## Explanation of Build Guidelines
- Multi-Page Application (MPA): The project employs a multi-page application architecture.
- Hash Removal from File Names: During the build, we ensure that file names no longer contain hash values.
- JavaScript File Naming Convention: JavaScript files adhere to a consistent naming convention corresponding to their respective HTML pages. For instance, the JavaScript file for about.html is named about.js.
- Non-Minified and Prettified HTML Output: Instead of the common practice of minifying HTML files, we retain non-minified versions, enhancing readability and maintenance by formatting them with Prettier. This preserves the original, legible formatting of HTML files.

## Question
I have made some configurations in Nuxt.js, successfully removing hash values from generated files. However, after running the build, I've noticed that there are no CSS and JS files in the 'dist' folder. Can someone help me troubleshoot this issue ?

## My Code
```javascript
  vite: {
    build: {
      rollupOptions: {
        output: {
          entryFileNames: `assets/[name].js`,
          chunkFileNames: `assets/[name].js`,

          assetFileNames: (assetInfo) => {
            if (assetInfo.name.endsWith('.css'))
              return 'assets/css/[name][extname]'

            if (/\.(png|jpe?g|gif|svg|webp)$/i.test(assetInfo.name))

              return 'assets/img/[name][extname]'

            if (/\.(woff|woff2|eot|ttf|otf)$/i.test(assetInfo.name))

              return 'assets/fonts/[name][extname]'

            if (/\.(mp4|webm|ogv)$/i.test(assetInfo.name))

              return 'assets/videos/[name][extname]'

            return 'assets/[name][extname]'
          },
        },
      },
    },
  },
```
## My Github Demo Repo
https://github.com/Michael0520/nuxt3-test-deploy/tree/main

## ScreenShot
![CleanShot 2024-01-08 at 15 22 39@2x](https://github.com/nuxt/nuxt/assets/60827537/31101118-5c4b-4159-a5ba-4de56ecf3432)

**Comment by masyoudi** [comment]
i don't have full explaination to build css and js files in nuxt, a little bit that i know in the rollupOptions is you need to provide full path filename

```ts
import { useNuxt } from '@nuxt/kit';
import { posix } from 'node:path';

export default defineNuxtConfig({
  vite: {
    build: {
      rollupOptions: {
        output: {
          entryFileNames: () => posix.join(String(useNuxt().options.vite.build?.assetsDir), '[name].js'),
          chunkFileNames: () => posix.join(String(useNuxt().options.vite.build?.assetsDir), '[name].js'),
          assetFileNames: (assetInfo) => {
            const resolver = (filePath: string) => posix.join(String(useNuxt().options.vite.build?.assetsDir), filePath);
            const assetName = assetInfo.name ?? ''

            if (assetName.endsWith('.css')) {
              return resolver('/css/[name][extname]');
            }

            if (/\.(png|jpe?g|gif|svg|webp)$/i.test(assetName)) {
              return resolver('/img/[name][extname]');
            }

            if (/\.(woff|woff2|eot|ttf|otf)$/i.test(assetName)) {
              return resolver('/fonts/[name][extname]');
            }

            if (/\.(mp4|webm|ogv)$/i.test(assetName)) {
              return resolver('/videos/[name][extname]');
            }

            return resolver('[name][extname]');
          }
        }
      }
    }
  }
});

```

**Reply by Michael0520** [reply]
## Question
Also, may I ask if you have any ideas on how to convert index.js into the correct page name?

For example: about.html => about.js"

## Lastest version about dist file screenshot
![CleanShot 2024-01-09 at 03 19 52](https://github.com/nuxt/nuxt/assets/60827537/0cc9f3d3-da92-4e82-9e9c-7cc7362efb6e)

**Reply by masyoudi** [reply]
i don't have idea on it
